### PR TITLE
feat: add getting started page (#586)

### LINKED ISSUE #586: Investigate how to create a start page similar to vscode

### PR DESCRIPTION
**Pull request by lstocchi**
it resolves #586 

lesson-tekton

cc @sudhirverma

### CODE REVIEW

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by lstocchi** [comment]
@mohitsuman I tried to tweak the settings of my recording tool and also increased the IJ IDE font, this is the best quality i could achieve for the gifs, give it a look please. Thanks!!

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                        "in read-only mode</p></html>",
+                Collections.emptyList(),
+                gifTreeLesson
+        );
+
+        URL gifNewFuncLesson = getGeneralLessonGif("createPipeline.gif");
+        GettingStartedLesson createPipelineLesson = new GettingStartedLesson(
+                "Create new resource",
+                "<html><p>The plugin provides a smart yaml-code completion support which make it easy to create " +
+                        "any Tekton resource by using the IDE editor.</p>" +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:106]:
```suggestion
                        "any Tekton resource directly from the IDE.</p>" +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                        "<p>By executing the `New <resource>` action a new editor tab opens up with a snippet " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:107]:
```suggestion
                        "<p>By executing the `New <resource>` action, a new editor tab opens up with a snippet " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+                        "for that specific resource. Right-click on a Kind node (e.g Pipelines) in the tree and select " +
+                        "the New action (e.g `New pipeline`) in the context menu. " +
+                        "Once you have done editing it, click the Push action in the top toolbar or in the " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:110]:
```suggestion
                        "Once you are done with the editing, click the Push action in the top toolbar or in the " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                        "notification bar.</p></html>",
+                Collections.emptyList(),
+                gifNewFuncLesson
+        );
+
+        URL gifBuildFuncLesson = getGeneralLessonGif("startPipeline.gif");
+        GettingStartedLesson startPipelineLesson = new GettingStartedLesson(
+                "Start a pipeline/task",
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:118]:
```suggestion
                "Start a Pipeline/Task",
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+                "<html>" +
+                        "<p>After having created/edited your pipeline/task and successfully pushed it on cluster, you can " +
+                        "run it by executing the `Start Pipeline/Task` action. " +
+                        "A wizard will pop up leading you through the process of setting all inputs necessary to " +
+                        "get it started (params, workspaces, service accounts, ..). " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:123]:
```suggestion
                        "get it started (params, workspaces, service accounts). " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+                        "<p>N.B: By default if the pipeline/task has no inputs, it will get started directly " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:124]:
```suggestion
                        "<p>Note: By default, the pipeline/task gets started directly if it has no inputs. " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+                        "without the wizard popping up. You can change this behavior by updating the plugin's settings " +
+                        "(File -> Settings -> Tools -> Tekton Pipelines By Red Hat). </p>" +
+                        "<p>If you already run a pipeline/task earlier and wants to run it again with the same inputs " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:127]:
```suggestion
                        "<p>If you have already run a pipeline/task earlier and want to run it again with the same inputs " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+                        "you did last time, you can achieve it by executing the `Start Last Run` action from the " +
+                        "Pipelines/Tasks context menu nodes</p>" +
+                        "<p>Opposite, if you want to start a pipeline/task with the same inputs of a specific old run " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:130]:
```suggestion
                        "<p>And similarly, if you want to start a pipeline/task with the same inputs of a specific old run " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+                        "To do this, just execute the `Show Diagnostic Data` action available when " +
+                        "right clicking a pipelinerun or taskrun node.</p></html>",
+                Collections.emptyList(),
+                gifDeployFuncLesson
+        );
+
+        URL gifRunFuncLesson = getGeneralLessonGif("debugTekton.gif");
+        GettingStartedLesson debugLesson = new GettingStartedLesson(
+                "Debug in Tekton",
+                "<html><p>When a task fails its execution and looking at the logs it is not enough to identify " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:157]:
```suggestion
                "<html><p>When a task fails its execution and the logs are not enough to identify " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+                        "the issue, you can think at starting it in debug mode. This way the container where the taskrun " +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:158]:
```suggestion
                        "the issue, you can start it in debug mode. This way the container where the taskrun " +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+                debugLesson,
+                tektonHubLesson);
+    }
+
+    private GettingStartedGroupLessons buildTektonCodeCompletionLessons() {
+        URL gifTreeLesson = getCodeCompletionLessonGif("taskRef.gif");
+        GettingStartedLesson taskRefLesson = new GettingStartedLesson(
+                "List all tasks for taskRef field",
+                "<html><p>When creating or editing a pipeline step, you can make the plugin import all params, " +
+                        "workspaces and resources of a task without typing them by hand.</p>" +
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:201]:
```suggestion
                        "workspaces and resources of a task automatically.</p>" +
```

File: src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+                        "list of pipeline inputs that can be used within the step.</p>" +
+                        "<p>Also the when clause and the old conditions CRD are supported</p>" +
+                        "</html>",
+                Collections.emptyList(),
+                gifBuildFuncLesson
+        );
+
+        return new GettingStartedGroupLessons(
+                "Tekton Code Completion",
+                "Creating or editing a Tekton resource is made easier",
```
Comment (mohitsuman) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/ui/toolwindow/GettingStartedToolWindow.java:234]:
```suggestion
                "Creating or Editing a Tekton resource is made easier",
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]
Added comments around the wordings. The gifs looks good and clear.

**Comment by lstocchi** [comment]
@mohitsuman Done, Thanks!!

**Review by mohitsuman — APPROVED** [review]
LGTM

**Pull request merged**